includes/api/class-wc-admin-rest-admin-notes-controller.php

```diff
@@ -0,0 +1,304 @@
+<?php
+/**
+ * REST API Admin Notes controller
+ *
+ * Handles requests to the admin notes endpoint.
+ *
+ * @package WooCommerce Admin/API
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Admin Notes controller class.
+ *
+ * @package WooCommerce/API
+ * @extends WC_REST_CRUD_Controller
+ */
+class WC_Admin_REST_Admin_Notes_Controller extends WC_REST_CRUD_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'admin/notes';
+
+	/**
+	 * Register the routes for admin notes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace, '/' . $this->rest_base, array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace, '/' . $this->rest_base . '/(?P<id>[\d-]+)', array(
+				'args'   => array(
+					'id' => array(
+						'description' => __( 'Unique ID for the resource.', 'wc-admin' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get a single note.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item( $request ) {
+		$note = WC_Admin_Notes::get_note( $request->get_param( 'id' ) );
+		if ( is_wp_error( $note ) ) {
+			return $note;
+		}
+
+		$data = $note->get_data();
+		$data = $this->prepare_item_for_response( $data, $request );
+		$data = $this->prepare_response_for_collection( $data );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Get all notes.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$per_page = isset( $request['per_page'] ) ? intval( $request['per_page'] ) : 10;
+		if ( $per_page <= 0 ) {
+			$per_page = 1;
```

Comment:
Should this use the default of 10 if per_page is less than zero?

Reply:
Agreed! Fixed in https://github.com/woocommerce/wc-admin/pull/507/commits/931b45f71d97a3ea5ee15244c68355514e947d95